akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala

```diff
@@ -408,9 +408,9 @@ object Replicator {
   sealed trait DeleteResponse[A <: ReplicatedData] {
     def key: Key[A]
   }
-  final case class DeleteSuccess[A <: ReplicatedData](key: Key[A]) extends DeleteResponse[A]
-  final case class ReplicationDeleteFailure[A <: ReplicatedData](key: Key[A]) extends DeleteResponse[A]
-  final case class DataDeleted[A <: ReplicatedData](key: Key[A])
+  final case class DeleteSuccess[A <: ReplicatedData](key: Key[A], request: Option[Any] = None) extends DeleteResponse[A]
+  final case class ReplicationDeleteFailure[A <: ReplicatedData](key: Key[A], request: Option[Any] = None) extends DeleteResponse[A]
+  final case class DataDeleted[A <: ReplicatedData](key: Key[A], request: Option[Any] = None)
```

Comment:
You send a `Delete` message and get a `DeleteSuccess` back as reply. The feature is that you can pass a request object in the `Delete` message same object is passed in the reply. Perhaps you should add a tests to verify that it works?